test/unit/sync/branch.test.ts

```diff
@@ -269,6 +287,13 @@ describe("sync/branches", () => {
 			expect(scope).not.toBeDone();
 			nock.removeInterceptor(interceptor);
 		});
+
+		it("should reschedule message with delay if there is rate limit", async () => {
```

Comment:
Adding test to the branches test, because it is the only test which covers error handling part of the backfill logic.

It is on my list to restore PRs and Commits tests too